pkg/loop/internal/capabilities_registry.go

```diff
@@ -0,0 +1,318 @@
+	}, nil
+}
+
+func (c *capabilitiesRegistryServer) GetTrigger(ctx context.Context, request *pb.GetTriggerRequest) (*pb.GetTriggerReply, error) {
+	capability, err := c.impl.GetTrigger(ctx, request.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	id, _, err := c.ServeNew("someTrigger", func(s *grpc.Server) {
+		pb.RegisterTriggerExecutableServer(s, &triggerExecutableServer{impl: capability})
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetTriggerReply{
+		CapabilityID: id,
+	}, nil
+}
+
+func (c *capabilitiesRegistryServer) GetAction(ctx context.Context, request *pb.GetActionRequest) (*pb.GetActionReply, error) {
+	capability, err := c.impl.GetAction(ctx, request.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	id, _, err := c.ServeNew("someAction", func(s *grpc.Server) {
+		pb.RegisterCallbackExecutableServer(s, &callbackExecutableServer{impl: capability})
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetActionReply{
+		CapabilityID: id,
+	}, nil
+
+}
+
+func (c *capabilitiesRegistryServer) GetConsensus(ctx context.Context, request *pb.GetConsensusRequest) (*pb.GetConsensusReply, error) {
+	capability, err := c.impl.GetConsensus(ctx, request.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	id, _, err := c.ServeNew("someConsensus", func(s *grpc.Server) {
+		pb.RegisterCallbackExecutableServer(s, &callbackExecutableServer{impl: capability})
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetConsensusReply{
+		CapabilityID: id,
+	}, nil
+}
+
+func (c *capabilitiesRegistryServer) GetTarget(ctx context.Context, request *pb.GetTargetRequest) (*pb.GetTargetReply, error) {
+	capability, err := c.impl.GetTarget(ctx, request.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	id, _, err := c.ServeNew("someTarget", func(s *grpc.Server) {
+		pb.RegisterCallbackExecutableServer(s, &callbackExecutableServer{impl: capability})
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetTargetReply{
+		CapabilityID: id,
+	}, nil
+}
+
+func (c *capabilitiesRegistryServer) List(ctx context.Context, _ *emptypb.Empty) (*pb.ListReply, error) {
+	capabilities, err := c.impl.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	reply := &pb.ListReply{}
+
+	var resources []Resource
+	for _, cap := range capabilities {
+		id, res, err := c.ServeNew("List", func(s *grpc.Server) {
+			pb.RegisterBaseCapabilityServer(s, &baseCapabilityServer{impl: cap})
+		})
+		if err != nil {
+			c.CloseAll(resources...)
+			return nil, err
+		}
+		resources = append(resources, res)
+		reply.CapabilityID = append(reply.CapabilityID, id)
+	}
+
+	return reply, nil
+}
+
+func (c *capabilitiesRegistryServer) Add(ctx context.Context, request *pb.AddRequest) (*emptypb.Empty, error) {
+	conn, err := c.Dial(request.CapabilityID)
+	if err != nil {
+		return &emptypb.Empty{}, ErrConnDial{Name: "Add", ID: request.CapabilityID, Err: err}
+	}
+	client := newBaseCapabilityClient(c.BrokerExt, conn)
```

Comment:
You need to check the execution type and wrap it in the correct underlying client type.